plugins/modules/azure_rm_adgroup.py

```diff
@@ -0,0 +1,409 @@
+        display_name: "Group-Name"
+        mail_nickname: "Group-Mail-Nickname"
+        state: 'present'
+        absent_members:
+          - "{{ ad_object_1_object_id }}"
+
+    - name: Ensure Users are Members of a Group using object_id
+      azure.azcollection.azure_rm_adgroup:
+        tenant: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        object_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        state: 'present'
+        absent_members:
+          - "{{ ad_object_1_object_id }}"
+
+    - name: Ensure Users are Owners of a Group using display_name and mail_nickname
+      azure.azcollection.azure_rm_adgroup:
+        tenant: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        display_name: "Group-Name"
+        mail_nickname: "Group-Mail-Nickname"
+        state: 'present'
+        present_owners:
+          - "https://graph.windows.net/{{ tenant_id }}/directoryObjects/{{ ad_object_1_object_id }}"
+          - "https://graph.windows.net/{{ tenant_id }}/directoryObjects/{{ ad_object_2_object_id }}"
+
+    - name: Ensure Users are Owners of a Group using object_id
+      azure.azcollection.azure_rm_adgroup:
+        tenant: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        object_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        state: 'present'
+        present_owners:
+          - "https://graph.windows.net/{{ ad_object_1_tenant_id }}/directoryObjects/{{ ad_object_1_object_id }}"
+          - "https://graph.windows.net/{{ ad_object_2_tenant_id }}/directoryObjects/{{ ad_object_2_object_id }}"
+
+    - name: Ensure Users are not Owners of a Group using display_name and mail_nickname
+      azure.azcollection.azure_rm_adgroup:
+        tenant: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        display_name: "Group-Name"
+        mail_nickname: "Group-Mail-Nickname"
+        state: 'present'
+        absent_owners:
+          - "{{ ad_object_1_object_id }}"
+          - "{{ ad_object_2_object_id }}"
+
+    - name: Ensure Users are Owners of a Group using object_id
+      azure.azcollection.azure_rm_adgroup:
+        tenant: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        object_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        state: 'present'
+        absent_owners:
+          - "{{ ad_object_1_object_id }}"
+          - "{{ ad_object_2_object_id }}"
+
+'''
+
+RETURN = '''
+'''
+
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common_ext import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.graphrbac.models import GraphErrorException
+    from azure.graphrbac.models import GroupCreateParameters
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMADGroup(AzureRMModuleBase):
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            object_id=dict(type='str'),
+            display_name=dict(type='str'),
+            mail_nickname=dict(type='str'),
+            present_members=dict(type='list', elements='str'),
```

Comment:
can we only have one members field here? Using the state present or absent to indicate the expected state of the listed members. And I also think if members field exists, we can assert that the ad group also exists.

Reply:
@haiyuazhang My concern with that is it might not be clear to the end-user how the module should behave. If we want users listed to be absent from a group but want the group to stay then we could say that passing any vars that aren't identifiers ( ie members and owners lists ) will have state applied not the group itself. From my experience with ansible, I would initially expect that the state always applies to the resource and expect those other fields to be ignored if absent was passed for the state. I think the separate lists make it more intuitive for the end-user but I have no issues updating the module if we want it to be more compact. Let me know your thoughts.

Reply:
You're right, Seems that I've made a stupid comment. As you put, If the expected state is absent,  We can't tell  if the expected state "absent" should apply to the whole group, means delete the group, or apply to the members in the group. 

But we can still treat the member list as part of the group state without explicitly specifying the absent members and present members . So to add members to or delete members from the group, We can always set the state to 'present', just modify the member list to reflect the changes you want to make. 

Another work around is to add an ad_rm_groupmember submodule to mange membership in the ad group, the same way as terraform azure ad provider does ([link](https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/resources/group_member)).  With the member sub module, It will be easier to do single member add/delete.

Reply:
Between the two workarounds, I think the better option is adding a group member module ( and a separate owner module as well). 

The issue I have with the first option (combining the fields in this module) is the need for the end-user to always know the status of all members of a group and not only the subset the user is concerned with. That's why I opted for the separate fields because it allows the user to only be concerned with the specific users they know should and should not be in the group; leaving all other users in the group alone. 

I think this same funcionality can be reached using the submodule approach, the only negative I can see really is adding another step to the provisioning/updating of a group. If we want to change to this route it shouldn't take that long to extract the logic out. Depending on the next planned promotion for release I can get these updates done within a couple of days of making a decision if we think this would be able to make it in by then.

Reply:
@coleneubauer  would you mind resolving the conflict: https://github.com/ansible-collections/azure/pull/423/conflicts ?
After that, I'll merge the PR.

Reply:
@haiyuazhang Resolved the conflict. Let me know if you need anything else, thanks